drivers/at86rf2xx/at86rf2xx.c

```diff
@@ -140,13 +289,25 @@ size_t at86rf2xx_send(at86rf2xx_t *dev, const uint8_t *data, size_t len)
 
 void at86rf2xx_tx_prepare(at86rf2xx_t *dev)
 {
-    uint8_t state;
+    uint8_t state = 0;
 
     dev->pending_tx++;
-    state = at86rf2xx_set_state(dev, AT86RF2XX_STATE_TX_ARET_ON);
-    if (state != AT86RF2XX_STATE_TX_ARET_ON) {
-        dev->idle_state = state;
+//    state = at86rf2xx_set_state(dev, AT86RF2XX_STATE_TX_ARET_ON);
+//    if (state != AT86RF2XX_STATE_TX_ARET_ON) {
+//        dev->idle_state = state;
+//    }
+
+    while (state != AT86RF2XX_STATE_TX_ARET_ON) {
```

Comment:
This change is not obvious what it fixes, what was broken before?
It should probably go in a separate commit.